projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -657,9 +674,15 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
     @ViewChild('readonlyInput', { read: ElementRef })
     protected readonlyInput: ElementRef;
 
-    /*
-    * @hidden
-    */
+     /* @hidden */
+    @ViewChild('editableInput', { read: IgxInputDirective })
+    protected editableInputDirective: IgxInputDirective;
+
+    /* @hidden */
+    @ViewChild('readonlyInput', { read: IgxInputDirective })
+    protected readonlyInputDirective: IgxInputDirective;
```

Comment:
Curious - would it also work if you just request `@ViewChild(IgxInputDirective)` and get the one from whichever template is active?